### DISCUSSION TITLE
How to increment the number in map by add

### DISCUSSION THREAD

**Discussion by Derek-X-Wang** [tywalch/electrodb#214] [Q&A] [answered]
Hi @tywalch,

Is there any way to increment the number inside of a map object?

For example

```typescript
export const UserEntity = new Entity(
  {
    model: {
      version: "1",
      entity: "User",
      service: "Demo",
    },
    attributes: {
      userId: {
        type: "string",
        required: true,
      },
      statistics: {
        type: "map",
        properties: {
          storeCount: {
            type: "number",
          },
          orderCount: {
            type: "number",
          },
        },
      },
      createdAt: {
        type: "number",
        default: () => Date.now(),
        // cannot be modified after created
        readOnly: true
      },
      updatedAt: {
        type: "number",
        // watch for changes to any attribute
        watch: "*",
        // set current timestamp when updated
        set: () => Date.now(),
        readOnly: true
      }
    },
    indexes: {
      user: {
        collection: "user",
        pk: {
          field: "pk",
          composite: ["userId"],
        },
        sk: {
          field: "sk",
          composite: [],
        },
      },
    },
  },
  Dynamo.Configuration
);
```

and I want to increment `storeCount` in `statistics` map object

```typescript
await UserEntity.update({
    userId,
  })
    .add({
      "statistics.storeCount": by, // something like this. not sure if electrodb has this feature.
    })
    .go();
```

**Comment by tywalch** [comment, marked as answer]
You can use the `.data()` method for this:

```typescript
UserEntity.update({ userId })
  .data((attr, op) => {
    op.add(attr.statistics.storeCount, 1)
  })
  .go();
```

[Full example in playground](https://electrodb.fun/?ssl=63&ssc=1&pln=64&pc=1#code/PQKgBAsg9gJgpgGzARwK5wE4Es4GcA0YuccYGeqCALgUQBYCG5YA7llXWAGbZwB2MXGBDAAUKKwBbAA5QMVMAG8wAUT5V2AT0IBlTADcsAY1IBfbhiiSwAIkRwjVSzABGNgNzijUPrgVUGFwRSAF5bTShUDAB9AKC4aL4GSTgPcTgAD1l5MG9fBQBVYgw1DSpNMDC+OBZVdS0AClEwJWaWsElYRAAuVvb2-UxcLB9emwBGG3w2-v4yzTGizCmZ9uLDEzGAEThOlf7Taf6GKicsF1QqPF7FVZbUYoBJGBu79vLpODG-bD4Ac32-Xa5DQWHILzATnQRyBhzefhOWD8xlwryB700nzGkgY0kB6Oklk+8hwqL66LWVDkcAAwpF1GiKf0Pl9bHxUJIXMsYUy4UyWnJ4Bg6agGeT+ZDMaybOzOdy3gcebClS0+UCjOQTnAYABBKiM9EssayrkYfFA+BcBiUfVgBoASkqAD4wFstQA6PhQFgOlXtYDAXIMPhehRcjqwLBcHAwMAMLhXDC5TVXGAKshwBgwADyfAQC0hGHQbzV-VQ0hgWt1ttuTKNbI5pvN-QDrBORk4XDkuUY-zwkKgcb4FROZwuV3TLHbdDGIGb-sDxAURii5HUkKkeACMlYdH4YHLldT6aXvQdztdHq9PvtfpampzeYLULgJdWpbAWAEmWu4vaD0wA11SgBBgkcEY+DGACzTvMBpAAayA9Fo0QCEbAQ+d+m8GQoGGK5egAbRsaDnhsABdWCP36XBEL-ZCcAQNCaMw9psNkPDWQIij0yosAPzVNVFDiYJTFEe1PFEPI-APJ5YzCAByQIjHkzwlhKepyndQ8tQaZQSNjUx7Tad0jwYBoGlHDBCCgaRHRCF1a3aGz3SzGALNODB3QRDRkSMXAvKpcgRXUQhxiM1VwrAd0-igB13CAA)

**Reply by Derek-X-Wang** [reply]
Thank you so much!
